---
title: "A01_afROSE_H3k27ac_Edit_stiched"
author:
- name:  Dr. Julio Domingo Cordero Henriquez
  affiliation: 
  - Department of Cardiovascular Genomics and Epigenomics, European Center for Angioscience (ECAS), Medical Faculty Mannheim, Heidelberg University, Mannheim, Germany
output:
  BiocStyle::html_document:
    toc: true
vignette: >
  % \VignetteIndexEntry{01Script: }
  % \VignetteEngine{knitr::rmarkdown}
---
```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
```


Mailto: julio.cordero@medma.uni-heidelberg.de<br />

# **Objective**
-To generate a Boxplots from Pause Index of output files from GitHub repository code **(https://github.com/MiMiroot/PIC)**.<br /><br />



# Summary of figures produced by this script.
```{r,fig.dim = c(13, 8), results="asis", align='center',out.width="150%", echo=F}
library(knitr)
#knitr::include_graphics("./BROAD_SUMMA_H3K4me3.png")
```   

```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=F}
library(future)
plan("sequential", workers = 16)
```


```{r,message=FALSE,echo=FALSE,warning=FALSE}
library(openxlsx)
library(data.table)
library(edgeR)
library(dplyr)
library("readxl")
library(doParallel)
library(BiocParallel)
library(gam)
library(foreach)
library(ggplot2)
library("RcppArmadillo")
library(DESeq2)
library(RColorBrewer)
library(Rcpp)
library(clusterProfiler)
library(org.Hs.eg.db)
library(plyr)
library(gplots)
library(EnrichedHeatmap)
library(ComplexHeatmap)
library(tidyr)
library(circlize)
library(stringr)
library("purrr")
library("dplyr")                                    # Load dplyr package
library("plyr")                                     # Load plyr package
library("readr")  
library(edgeR)
library(ggplot2)
library(rstatix)
library(ggpubr)
```


```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=F}
workdir = "./"
setwd(workdir)

PROJECT="A01_afROSE_H3k27ac_Edit_stiched"
PTHA11="./03OUTPUT/"
dir.create(PTHA11)
PTHA=paste(PTHA11,PROJECT,"/",sep="")
PORT=paste(PTHA11,PROJECT,"/",sep="")
dir.create(PTHA)
dir.create(PORT)
PEAK1="01Ctr"
PEAK2="02R20"

meta2 = "./"
mat_m1b = data.frame(list.files(path=paste(meta2,sep=""), pattern="*_s_nChrMunionpeaks_Al_for_forR_bedto.bed", full.names=F))
mat_m22 = data.frame(gsub("*_s_nChrMunionpeaks_Al_for_forR_bedto.bed", "", mat_m1b[,1]))

G4<- data.frame(str_split_fixed(mat_m22[,1],"_", 6))

mat_m22$Sample<- G4[,3]
#name2<- as.character(t(mat_m22[,2]))


# mat_v108 = read.delim(WORK1,header=F,sep="\t",dec=".")
#mat_v108<- subset(mat_v108,select=c("V4","V8"))
#colnames(mat_v108)<- c("gene_id","SYMBOL")

#c2 = "../01DATA/gencode.v45.basic_more1kb.txt"

#cat("Current file name is:",c2,"\n")
#V45 = read.delim(c2,header=T,sep="\t",dec=".")


```

#### this is a MARVELOUS WAY to integrate a lot of Quant files from RNA-seq  ######
## THANK JESUS ####
## Load annotations

```{r}
WORK2="annotation/"
mat_m1b1 = data.frame(list.files(path=paste(WORK2,sep=""), pattern="*_ALL_v_MACS3bro_200_001_anno.bed", full.names=T))
DE_LIST<- list()
DE_LIST2<- list()
NAME_GN5<- c("ctr_RO","st_RO","en_RO","ID_RO","V5","RO_size","Normalized_Tag_Count","ID_RO2","isSE","ctr_p","st_p","en_p" ,"Gene.Name","Nearest.PromoterID","Annotation","Distance.to.TSS")
 for(i in 1:length(mat_m22$Sample)){
 MO<- mat_m22[i,1]
 print(MO)
 
 mat_1 = read.delim(mat_m1b1[i,1],header=F,sep="\t",dec=".")
colnames(mat_1)<- NAME_GN5
mat_1$CELL<- G4[i,1]
mat_1$HM<- G4[i,2]
mat_1$Sample<- G4[i,3]
mat_1$ENH4[mat_1$isSE==1]<- "01SE"
mat_1$ENH4[mat_1$isSE==0]<- "02TYE"
# DE_LIST[[i]]<- subset(mat_3,select=c(NAME_PI2))
  DE_LIST2[[i]]<- mat_1
# DE_LIST[[i]]<- subset(mat_3,select=c(NAME_PI2))

 }
NAME_GBro<- c("ctr_RO","st_RO","en_RO",  "PEAK_ID")
GN5ano<- do.call(rbind, DE_LIST2)


             
             
GN5ano$PEAK_ID<- paste(GN5ano$ctr_RO,GN5ano$st_RO,GN5ano$en_RO,GN5ano$CELL,GN5ano$Sample,GN5ano$ENH4,GN5ano$Gene.Name,sep="_")

GN5ano_u <-GN5ano[!duplicated(GN5ano[,c("PEAK_ID")]),]

table(GN5ano_u$Sample,GN5ano_u$ENH4)
write.table(GN5ano,file=paste(PTHA,PROJECT,"_fromR.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
write.table(GN5ano,file=paste(PTHA,PROJECT,"_fromR_w_TITLE.txt",sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)

```

## for IGV
```{r,echo=FALSE}
#GN5ano$CELL
NAME_GBro2<- plyr::count(GN5ano,c("Sample"))
NAME_GBro3<- plyr::count(GN5ano,c("ENH4"))

DE_LIST2<- list()
 for(i in 1:length(NAME_GBro2$Sample)){
    for(B in 1:length(NAME_GBro3$ENH4)){
SA<- as.character(NAME_GBro2[i,"Sample"])
ENH<- as.character(NAME_GBro3[B,"ENH4"])
MO33<- paste("01MLgH3k27ac", SA,ENH,sep="_")
 print(MO33) 
DE_LIST2[[MO33]]<- subset(GN5ano,Sample==SA &ENH4==ENH,  select=c(NAME_GBro))
write.table(DE_LIST2[[MO33]],file=paste(PTHA,MO33,"_fromRIGV.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
}}
NAME_GBro4<- plyr::count(GN5ano,c("Sample","ENH4"))
GN5ano_u1 <-GN5ano[!duplicated(GN5ano[,c("ID_RO","Sample")]),]
MA6<- plyr::count(GN5ano_u1,c("Sample","ENH4"))
NAME_SA<- plyr::count(GN5ano_u1,c("Sample"))
```


### Peaks after rmd

```{r,echo=FALSE}
library(Hmisc)
library(tidyr)

df.list<- list()
for(i in 1:length(NAME_SA$Sample)) {
  SA<- as.character(NAME_SA[i,1])
df.list[[i]] <- subset(GN5ano_u1,Sample==SA )


df.list[[i]] <- df.list[[i]][order(df.list[[i]]$Normalized_Tag_Count,decreasing=F),]

  df.list[[i]]$XLA<- 1:nrow(df.list[[i]])
df.list[[i]]$ID <-NAME_SA[i,1]
#df.list[[i]]$ENH4[df.list[[i]]$isSuper==1]<- "01SE"
#df.list[[i]]$ENH4[df.list[[i]]$isSuper==0]<- "02TYE"
}


CHECK02<- (df.list[[2]])
#CHECK03<- (df.list[[3]])


#k4m3_45dpi_merge_v_INP_MACS2_bro_200_001_peak_17001
GN5a<- do.call(rbind, df.list)
#CHECK1<- subset(GN5a, Var2=="01PROM(<2kb)" | Var2=="02Genebody(>2kb)" | Var2=="03Interg" )
 CO22S<- c("darkgrey","#01665e","darkgreen","#f16913","#df65b0","#8c510a")

```




```{r}
###option fill is giving the result in percentage and stack in total values
XL="Enhancer ranked by H3k27ac signal"
YL="Peak distribut (%)"
YL2="Norm H3k27ac signal"
L1<- min(GN5a$XLA)
L2<- round(max(GN5a$XLA)/2,digits = 0)
L3<- round(max(GN5a$XLA)*0.75,digits = 0)
L4<- round(max(GN5a$XLA),digits = 0)
S4<- round(max(GN5a$Normalized_Tag_Count),digits = 0)
S3<- S4/2
S4
LL=12
Si=2
MAX=10000
PTHA
BASIC_COL="black"
SAM<- count(GN5a$ID)
GR<- plyr::count(GN5a,c("CELL"))
GR1<- as.character(GR[1,1])

F1<- ggplot(GN5a) + geom_line(aes(XLA,Normalized_Tag_Count, colour=ID),size=2) +
  scale_color_manual(name=GR1,values=CO22S)+ylim(0,S3)+
   theme_bw() +
    theme(strip.placement = "outside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,size=LL, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=Si),axis.text.y = element_text(size=LL, face="bold")) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+scale_x_continuous(breaks = seq(0, 21000, by = 4000 ))
 library(ggforce)

    print(F1)
  F2<- ggplot(GN5a) + geom_line(aes(XLA,(Normalized_Tag_Count), colour=ID),size=2) +
  scale_color_manual(name=GR1,values=CO22S)+
   theme_bw() +
    theme(strip.placement = "outside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,size=LL, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=Si),axis.text.y = element_text(size=LL, face="bold")) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+scale_x_continuous(breaks = c( L1,L3,L4 ))+
  facet_zoom(ylim = c(0, S3), zoom.data = ifelse(a <= S2, NA, FALSE))
  
  F3<- ggplot(GN5a) + geom_line(aes(XLA,Normalized_Tag_Count, colour=ID),size=2) +
  scale_color_manual(name=GR1,values=CO22S)+
   theme_bw() +
    theme(strip.placement = "outside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,size=LL, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=Si),axis.text.y = element_text(size=LL, face="bold")) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+scale_x_continuous(breaks = seq(L1, L4, by = 3999)) +scale_y_continuous(limits = c(0, S4), breaks = seq(0, S4, by = S4/2))
  
  #+ scale_y_continuous(limits = c(0, 2000), breaks = seq(0, 2000, by = 200))


  print(F2)
    print(F3)


arrange6a <- ggarrange(F1, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")
arrange6a2 <- ggarrange(F2, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")

pdf(file=paste(PTHA,PROJECT,".pdf", sep=""),width=5, height=4)  
{
print(arrange6a)
print(arrange6a2)
  dev.off()}

write.xlsx(GN5a,file=paste(PORT,"00", "_",PROJECT, "_GN5a_MATRIX.xlsx",sep=""),overwrite = T)


```


## calculate P-value of the enrichment of the K27ac

```{r}
table(GN5a$RO_sizeENH4)
mat_RPKM1_5_c<- subset(GN5a,ENH4=="01SE")
P11t2CH<-  ggplot(mat_RPKM1_5_c) + geom_line(aes(XLA,(Normalized_Tag_Count), colour=ID),size=2) +  scale_color_manual(name=GR1,values=CO22S)+ theme_bw() +  theme(strip.placement = "outside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,size=LL, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=Si),axis.text.y = element_text(size=LL, face="bold")) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+scale_x_continuous(breaks = c( L1,L3,L4 ))

print(P11t2CH)
library(rstatix)
library(ggpubr)
stat.test5 <- mat_RPKM1_5_c %>%
#  group_by(DE2) %>%
    wilcox_test(Normalized_Tag_Count ~ ID) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj")
stat.test5
stat.test5 <- stat.test5 %>%
  add_xy_position(x = "Sample", dodge = 0.8)
YPOS5<- as.numeric((stat.test5$y.position )/3)
SI=10  
stat.test6<- stat.test5
stat.test6$log10p<- log10(stat.test6$p)*(-1)
stat.test6$log10fdr<- log10(stat.test6$p.adj)*(-1)
stat.test6$CELL<- paste(stat.test6$DE2,stat.test6$group1,stat.test6$group2,sep="_")
write.xlsx(stat.test6,file=paste(PORT,"00", "_",PROJECT, "_Pvalues.xlsx",sep=""),overwrite = T)

```


```{r}
phu <-ggplot(GN5a, aes(x=log10(RO_size),fill = ID)) +   geom_density( alpha=0.3)+scale_fill_manual(values=c("darkgrey","darkgreen"))+
  theme_bw()+facet_wrap(~ENH4, ncol = 2,strip.position="top",scales="free")+
   labs(title=paste(PROJECT))
print(phu)
pdf(file=paste(PTHA,PROJECT,"_DENSY_by_ENH.pdf", sep=""),width=5, height=4)  
print(phu)

dev.off()
```
### check for MLg cells Peak Size of H3K4me3 in Ctr and Mir9-LOF
```{r,echo=FALSE}
GN5a$Sample
GN5a$length<- GN5a$RO_size
### To improve the annotation to genename
MAR_TOTAL<- list()
GN5e1b<- list()

  NAME_SA<- plyr::count(GN5ano_u1,c("Sample"))
NAME_BRO<- plyr::count(GN5ano_u1,c("ENH4"))


#for(i in 1:length(NAME_GE$Var1)) {
  for(B in 1:length(NAME_BRO$ENH4)) {
  #  GENO<- as.character(NAME_GE[i,1])
    BRO<- as.character(NAME_BRO[B,1])
    CC<- paste(BRO,sep="_")
    print(CC)
  
GN5e1b[[CC]]<- subset(GN5a, ENH4==BRO)

FIL<- data.matrix(summary(GN5e1b[[CC]]$length))
FIL2<-as.numeric(round(FIL[2,1],digits = 1))

############ select the peak Wide from each data 
FIL5<-as.numeric(round(FIL[5,1],digits = 1))
FIL3<-as.numeric(round(FIL[3,1],digits = 1))

FIL5b<-as.numeric(round(FIL5/1000,digits = 1))
FIL3b<-as.numeric(round(FIL3/1000,digits = 1))

GN5e1b[[CC]]$CLU[GN5e1b[[CC]]$length >=FIL5] <- paste("01WID",">",FIL5b,"kb",sep=" ")
GN5e1b[[CC]]$CLU[GN5e1b[[CC]]$length <FIL5 & GN5e1b[[CC]]$length>= FIL3 ] <- paste("02ME_","<",FIL5b,"kb",">",FIL3b,"kb",sep="")
GN5e1b[[CC]]$CLU[GN5e1b[[CC]]$length < FIL3] <- paste("03SMAL_","<",FIL3b,"kb",sep="")

GN5e1b[[CC]]$CLU2[GN5e1b[[CC]]$length >=FIL5] <- "01WID"
GN5e1b[[CC]]$CLU2[GN5e1b[[CC]]$length <FIL5 & GN5e1b[[CC]]$length>= FIL3 ] <- "02ME"
GN5e1b[[CC]]$CLU2[GN5e1b[[CC]]$length < FIL3] <- "03SMAL"
  } 
##write.table(GN5e2,file=paste(PTHA,PROJECT,"PEAK_fromR2.txt",sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)
brewer.pal(n = 11, name = "Dark2")
MI=6
COL<- "black" 
#table(GN5e2$Var2)
```

```{r}
GN5e1b_WIDE<- do.call(rbind, GN5e1b)
write.xlsx(GN5e1b_WIDE,file=paste(PORT,"00", "_",PROJECT, "_MAT_fPLOT.xlsx",sep=""),overwrite = T)
MA6_WIDE<- plyr::count(GN5e1b_WIDE,c( "ENH4","Sample","CLU2","CLU"))

write.table(GN5e1b_WIDE,file=paste(PTHA,PROJECT,"_wWIDE_n_TITLE.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
MA6_WIDE$CELL<- "MLg"
```








##
```{r}
NAME33b<- c( "01c","02Lof")
NAME13b<- NAME33b
YL_EXP2="Zscore"
YL<- "Peak distribution (%)"
YL2<- "NÂ° of Peaks"
XL=" "
YL_EXP="Log2(meanRPKM+1)"
IP="H3k27ac"
#rm(p)
YL_EXP2="Zscore"
  LE=14
  SI=14
BASIC_COL="black"
PEAK="Peak group"

###option fill is giving the result in percentage and stack in total values


#CO22P <-  c("#4d004b","#8c6bb1","#9ebcda", "gray","darkblue", "black")
#  ['#ffffd9','#edf8b1','#c7e9b4','#7fcdbb','#41b6c4','#1d91c0','#225ea8','#0c2c84']
 # CO22P<- c('#225ea8','#41b6c4','#c7e9b4')
     CO22P<- c('mediumaquamarine','#FFB5C2','#E5E5E5')
  #  CO22P<- c('#3777FF','#41b6c4','#E5E5E5')
p11 <- ggplot(MA6_WIDE, aes(factor(Sample), freq, fill = factor(CLU2))) +
        geom_bar(position = "fill",stat = "identity") +
        guides(fill = guide_legend(reverse = F)) +
        labs(fill = PEAK)+
   theme_bw()+facet_wrap(~ENH4, ncol = 2,strip.position="top",scales="fixed")+   scale_fill_manual(values =CO22P) +
   theme(strip.placement = "outside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,size=SI , face="bold"),axis.text.y=element_text(angle =0,size=SI , face="bold"), panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL)+ scale_y_continuous(breaks=seq(0,25,0.5))



print(p11)

b11 <- ggplot(MA6_WIDE, aes(factor(Sample), freq, fill = factor(CLU2))) +       geom_bar(position = "stack",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +labs(fill = PEAK) +   theme_bw()+facet_wrap(~CELL+ENH4, ncol =2,strip.position="top",scales="free")+   scale_fill_manual(values =CO22P) + theme(strip.placement = "outside",strip.text.y.left =element_text(angle=0),axis.text.x=element_text(angle = 0,size=SI ,face="bold"),axis.text.y=element_text(angle = 0,size=SI , face="bold"), panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)

b112 <- ggplot(MA6_WIDE, aes(factor(Sample), freq, fill = factor(Sample))) +       geom_bar(position = "stack",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +labs(fill = PEAK) +   theme_bw()+facet_wrap(~CELL+ENH4, ncol =2,strip.position="top",scales="free")+   scale_fill_manual(values =CO22S) + theme(strip.placement = "outside",strip.text.y.left =element_text(angle=0),axis.text.x=element_text(angle = 0,size=SI ,face="bold"),axis.text.y=element_text(angle = 0,size=SI , face="bold"), panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)
  ##+coord_flip()+  to flip the plot


print(b11)
print(b112)
#print(b21)
TY1="01MLg_"
arrange6a1 <- ggarrange(p11, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")
arrange6a2 <- ggarrange(b11, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")
arrange6a22 <- ggarrange(b112, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")
pdf(file=paste(PTHA,TY1, PROJECT,"WID_SE_TYE", ".pdf", sep=""),width=5, height=4)  
print(arrange6a1)
print(arrange6a2)
print(arrange6a22)
dev.off()
b11212 <- ggplot(MA6_WIDE, aes(factor(Sample), freq, fill = factor(Sample))) +       geom_bar(position = "stack",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +labs(fill = PEAK) +   theme_bw()+facet_wrap(~CELL, ncol =1,strip.position="top",scales="free")+   scale_fill_manual(values =CO22S) + theme(strip.placement = "outside",strip.text.y.left =element_text(angle=0),axis.text.x=element_text(angle = 0,size=SI ,face="bold"),axis.text.y=element_text(angle = 0,size=SI , face="bold"), panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)
arrange6a222 <- ggarrange(b11212, ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="top")
pdf(file=paste(PTHA,"00_A", TY1, PROJECT,".pdf", sep=""),width=3, height=3)  
{
  print(arrange6a222)
dev.off()}
```


## Creat the Matrix of Percentage for MLg
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=T}

TT<- sum(MA6_WIDE$freq)
MA6_WIDE$ENH_WIDE<- paste(MA6_WIDE$Sample,MA6_WIDE$ENH4,sep="_")
NAME_GE<- data.frame(table(MA6_WIDE$ENH_WIDE))
NAME_SA<- data.frame(table(MA6_WIDE$Sample))

df.list3<- list()
for(i in 1:length(NAME_GE$Freq)) {
 
   for(A in 1:length(NAME_SA$Freq)) {
  GENO<- as.character(NAME_GE[i,1])
   SA<- as.character(NAME_SA[A,1])
   CC<- paste(GENO,SA,sep="_")
   print(CC)

  df.list3[[CC]] <- subset(MA6_WIDE, ENH_WIDE==GENO &Sample==SA)
  TT<- sum(df.list3[[CC]]$freq)
df.list3[[CC]]$PER<- c(df.list3[[CC]]$freq*100)/TT
  }
}
GN5_fish<- do.call(rbind, df.list3)
#NAME_BRO<- data.frame(table(MA6_WIDE$CLU2))


```

## Perform Fisher`s Exact test and ChiP square test from the percentage 
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=T}

TY1="01MLg_"
NAME_TEST<- c("freq","noCO2")
NAME_TEST_PER<- c("PER","noCO")
NAME_BRO<- data.frame(table(MA6_WIDE$CLU2))
NAME_ENH4<- data.frame(table(MA6_WIDE$ENH4))

MAR<- list()
MAT_for_TEST<- list()
i=1
B=1
for(i in 1:length(NAME_ENH4$Var1)) {
  for(B in 1:length(NAME_BRO$Var1)) {
  ENH<- as.character(NAME_ENH4[i,1])
   BRO<- as.character(NAME_BRO[B,1])
   CC<- paste(ENH,BRO,sep="_")
   print(CC)
#}}
  MAT_for_fish <- subset(GN5_fish, ENH4==ENH & CLU2==BRO )
  #    MAT_for_fish1 <- subset(GN5_fishA, ENH4=="03NARROW"&CLU2==BRO,select=c("BRO", "PER" ))

MAT_for_fish$noCO<- 100-MAT_for_fish$PER
SM<- sum(MAT_for_fish$freq)
MAT_for_fish$noCO2<- SM-MAT_for_fish$freq
MAT_for_TEST[[CC]]<- MAT_for_fish[1:2,NAME_TEST_PER]
MAT_for_TEST[[CC]]$COMBI<- CC
PVA_fisher<- fisher.test(MAT_for_TEST[[CC]][1:2])
 PVA_chip<- chisq.test(MAT_for_TEST[[CC]][1:2])


MAR[[CC]]<- data.frame(PVA_fisher$p.value)
MAR[[CC]]$AREA<- CC
MAR[[CC]]$Sam<- "Ctr_vs_Mir9_LOF"
MAR[[CC]]$method<- "Fisher_Exact_test"
MAR[[CC]]$log10P<- log10(MAR[[CC]][,1])*(-1)
MAR[[CC]]$pval_chisqtest<- PVA_chip$p.value
MAR[[CC]]$method2<- "chisqtest"
MAR[[CC]]$log10Pfisher<- log10(MAR[[CC]][,1])*(-1)
MAR[[CC]]$log10PChip<- log10(MAR[[CC]][,1])*(-1)

 }
}
GN5_fishMATR<- do.call(rbind, MAR)
GN5_fishMATR_MATX<- do.call(rbind, MAT_for_TEST)
GN5_fishMATR_MATX$SAM_COM<- rownames(GN5_fishMATR_MATX)

fdrs<-p.adjust(GN5_fishMATR$PVA_fisher.p.value, method="BH")
write.xlsx(GN5_fishMATR,file=paste(PTHA,TY1, PROJECT,"Pva_bar_PER",".xlsx", sep=""),rowNames=F,overwrite = T)
write.table(GN5_fishMATR,file=paste(PTHA,TY1, PROJECT,"Pva_bar_PER",".txt", sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)
write.xlsx(GN5_fishMATR_MATX,file=paste(PTHA,TY1, PROJECT,"Pva_bar_PER_MAT_FIG",".xlsx", sep=""),rowNames=F,overwrite = T)
```


## to cross with Mir9

```{r}
NAME_WIDE<- plyr::count(GN5e1b_WIDE,c("CLU2"))
NAME_ENH<- plyr::count(GN5e1b_WIDE,c("ENH4"))
 mat_ALL = read.delim("./04merge_MLg/01MLg_K27ac_merPE_v_fROSE_annot_BEDTOOLS_INDIV_PEAKS_WITH_ROSE_SE.bed",header=F,sep="\t",dec=".")
 NAME_SE_PEAKS<- paste("V",c(5:16),sep="")
mat_ALL2<- subset(mat_ALL,V23=="Ctr",select=c(NAME_SE_PEAKS,"V23") )
mat_ALL2$PEAK_ID2<- paste(mat_ALL2$V5,mat_ALL2$V6,mat_ALL2$V7,mat_ALL2$V23,sep="_")
GN5e1b_WIDE$PEAK_ID2<- paste(GN5e1b_WIDE$ctr_RO,GN5e1b_WIDE$st_RO,GN5e1b_WIDE$en_RO,GN5e1b_WIDE$Sample,sep="_")
GN5e1b_WIDE_Ctr<- subset(GN5e1b_WIDE,Sample=="Ctr")
colnames(mat_ALL2)[1]<- "ctr_RO1"


GN5e1b_WIDE2<- merge(mat_ALL2,GN5e1b_WIDE_Ctr,by="PEAK_ID2")
 NAME_SE_PEAKS22<-c(colnames(GN5e1b_WIDE2[,2:ncol(GN5e1b_WIDE2)]))

PTHA_2<- paste(PTHA,"toCrossBed_WID_A","/",sep="")
dir.create(PTHA_2)
PTHA_3<- paste(PTHA,"toCrossBed_WID_SEP","/",sep="")
dir.create(PTHA_3)
GN5e1b_WIDE2_PEAKS<- subset(GN5e1b_WIDE2,Sample=="Ctr",select=NAME_SE_PEAKS22)

write.table(GN5e1b_WIDE2_PEAKS,file=paste(PTHA_2,"01Mlg_ctr","merge_fROSE_WID_A.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)

for(i in 1:length(NAME_ENH$ENH4)) {
  for(B in 1:length(NAME_WIDE$CLU2)) {
    SE<- as.character(NAME_ENH[i,1])
    BRO<- as.character(NAME_WIDE[B,1])
    CC<- paste("01Mlg_ctr", BRO,SE,sep="_")
    print(CC)
  
A_MAT_SE<- subset(GN5e1b_WIDE2, CLU2==BRO &ENH4==SE,select=NAME_SE_PEAKS22 )

  A_MAT_SE <- A_MAT_SE[order(A_MAT_SE$ctr_RO1,A_MAT_SE$V7,A_MAT_SE$V6,decreasing=F),]

write.table(A_MAT_SE,file=paste(PTHA_3,CC,"_fromR.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
}}
```







```{r,echo=FALSE}

  #A_MAT_SE <- A_MAT_SE[A_MAT_SE$chr,A_MAT_SE$chr,A_MAT_SE$endchr,decreasing=F),]
  A_MAT_SE <- A_MAT_SE[order(A_MAT_SE$chr,A_MAT_SE$start,A_MAT_SE$end,decreasing=F),]

write.table(A_MAT_SE,file=paste(PTHA,PROJECT,"_fromR.txt",sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)
write.table(A_MAT_SE,file=paste(PTHA,PROJECT,"_fromR.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)

GN5e1b_WIDE<- do.call(rbind, GN5e1b)

MA66<- plyr::count(GN5e1b_WIDE,c("Sample","ENH4"))
write.table(GN5ano[,NAME_GBro],file=paste(PTHA,PROJECT,"_fromRIGV.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
```


